tool/ci/pkg/environment/environment.go

```diff
@@ -0,0 +1,301 @@
+		Client:           c.Client,
+		reviewers:        revs,
+		defaultReviewers: revs[""],
+		Metadata:         pr,
+	}, nil
+}
+
+// unmarshalReviewers converts the passed in string representing json object into a map
+func unmarshalReviewers(ctx context.Context, str string, client *github.Client) (map[string][]string, error) {
+	var hasDefaultReviewers bool
+	if str == "" {
+		return nil, trace.NotFound("reviewers not found")
+	}
+	m := make(map[string][]string)
+
+	err := json.Unmarshal([]byte(str), &m)
+	if err != nil {
+		return nil, err
+	}
+	for author, requiredReviewers := range m {
+		for _, reviewer := range requiredReviewers {
+			_, err := userExists(ctx, reviewer, client)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+		}
+		if author == "" {
+			hasDefaultReviewers = true
+			continue
+		}
+		_, err := userExists(ctx, author, client)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	if !hasDefaultReviewers {
+		return nil, trace.BadParameter("default reviewers are not set. set default reviewers with an empty string as a key")
+	}
+	return m, nil
+
+}
+
+// userExists checks if a user exists
+func userExists(ctx context.Context, userLogin string, client *github.Client) (*github.User, error) {
+	user, resp, err := client.Users.Get(ctx, userLogin)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return nil, trace.Wrap(err)
+	}
+	return user, nil
+}
+
+// GetReviewersForAuthor gets the required reviewers for the current user.
+func (e *PullRequestEnvironment) GetReviewersForAuthor(user string) []string {
+	value, ok := e.reviewers[user]
+	// author is external or does not have set reviewers
+	if !ok {
+		return e.defaultReviewers
+	}
+	return value
+}
+
+// IsInternal determines if an author is an internal contributor.
+func (e *PullRequestEnvironment) IsInternal(author string) bool {
+	_, ok := e.reviewers[author]
+	return ok
+}
+
+// GetMetadata gets the pull request metadata in the current context.
+func GetMetadata(path string) (*Metadata, error) {
+	var actionType action
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	defer file.Close()
+	body, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	err = json.Unmarshal(body, &actionType)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return getMetadata(body, actionType.Action)
+}
+
+func getMetadata(body []byte, action string) (*Metadata, error) {
+	var pr Metadata
+
+	switch action {
+	case ci.Synchronize:
+		var push PushEvent
+		err := json.Unmarshal(body, &push)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		pr, err = push.toMetadata()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	case ci.Assigned, ci.Opened, ci.Reopened, ci.Ready:
+		var pull PullRequestEvent
+		err := json.Unmarshal(body, &pull)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		pr, err = pull.toMetadata()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	default:
```

Comment:
The workflow files only trigger on `pull_request` and `pull_request_review` events and gets the event payload from the runner so that it is unlikely that the action will be something else, but I will create a default, thanks for catching that!